spec/consensus/reactor/reactor.md

```diff
@@ -0,0 +1,355 @@
+
+Tendermint BFT also assumes that this property is used to provide **Gossip Communication**:
+
+> **Gossip communication**: (i)If a correct process $p$ sends some message $m$ at time $t$, all correct processes will receive $m$ before $\text{max} \{t,\text{GST}\} + \Delta$.
+Furthermore, (ii)if a correct process $p$ receives some message $m$ at time $t$, all correct processes will receive $m$ before $\text{max}\{t,\text{GST}\} + \Delta$.
+
+Because Gossip Communication requires even messages sent before GST to be reliably delivered between correct processes ($t$ may happen before GST in (i)) and because GST could take arbitrarily long to arrive, in practice, implementing this property would require an unbounded message buffer.
+
+However, while Gossip Communication is a sufficient condition for Tendermint BFT to terminate, it is not strictly necessary that all messages from correct processes reach all other correct processes.
+What is required is for either the message to be delivered or that, eventually, some newer message, with information that supersedes that in the first message, to be timely delivered.
+In Tendermint BFT, this property is seen, for example, when nodes ignore proposal messages from prior rounds.
+
+> **Supersession**: We say that a message $lhs$ supersedes a message $rhs$ if after receiving $lhs$ a process would make at least as much progress as it would by receiving $rhs$ and we note is as follows: $\text{lhs} \overset{s}{>} \text{rhs}$.   
+> **TODO**: Better definition.
+
+It seems reasonable, therefore, to formalize the requirements of Tendermint in terms of a weaker, *best-effort* communication guarantees and to combine them with GST outside of the communication layer, to ensure eventual termination.
+
+> **Note**   
+> It is also assumed that, after GST, timeouts do not expire precociously.
+
+
+
+
+
+# Part 2: Specifications
+## The Consensus State Layer - State
+
+### Provides to Applications and other Reactors
+
+Communication of the Consensus Reactor with layers above (applications) and on to the side (other reactors) is covered by the [ABCI](../../abci/) specification.
+
+### Provides to the Communication Layer
+
+A message supersession operator:
+
+`SSS(lhs,rhs)` (`lhs` supersedes `rhs`) is true if and only if $\text{lhs} \overset{s}{>} \text{rhs}$
+
+### Requires from the Communication Layer
+
+Messages received by a node either change the node's State or are discarded. 
+If they change the State, even if by just getting added to a set of received messages, then the original message need not be forwarded. 
+Instead, a new message, with the same contents, is generated from the State of the node. Why does this matter? Because it let's us think in terms of messages being sent to neighbors only, not forwarded on the overlay.
```

Comment:
I agree that this allows us to think about the neighbors only. At the same time it introduces a level of disconnection from the properties in the arXiv paper